bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreClientImpl.java

```diff
@@ -0,0 +1,660 @@
+        new ExternalCatalogTableOptions()
+            .setStorageDescriptor(table.getExternalCatalogTableOptions().getStorageDescriptor())
+            .setConnectionId(table.getExternalCatalogTableOptions().getConnectionId())
+            .setParameters(table.getExternalCatalogTableOptions().getParameters());
+    Table updatedTable =
+        new Table()
+            .setExternalCatalogTableOptions(newExternalCatalogTableOptions)
+            // Must set the schema as null for using schema auto-detect.
+            .setSchema(Data.nullOf(TableSchema.class));
+
+    Table response = null;
+    try {
+      response =
+          BigQueryRetryHelper.runWithRetries(
+              () -> internalUpdate(tableReference, updatedTable, table.getEtag()),
+              bigqueryOptions.getRetrySettings(),
+              BIGQUERY_EXCEPTION_HANDLER,
+              bigqueryOptions.getClock(),
+              DEFAULT_RETRY_CONFIG);
+    } catch (BigQueryRetryHelper.BigQueryRetryHelperException e) {
+      handleBigQueryRetryException(e);
+    }
+
+    return response;
+  }
+
+  private Table internalUpdate(TableReference tableReference, Table table, String etag) {
+    try {
+      HttpResponse response =
+          client
+              .tables()
+              .patch(
+                  tableReference.getProjectId(),
+                  tableReference.getDatasetId(),
+                  tableReference.getTableId(),
+                  table)
+              .setRequestHeaders(new HttpHeaders().setIfMatch(etag))
+              .executeUnparsed();
+
+      if (response.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+        String responseString = response.parseAsString();
+        if (responseString.toLowerCase(Locale.ENGLISH).contains("not found: connection")) {
+          throw new BadRequestException("%s", responseString);
+        }
+
+        throw new NoSuchTableException("%s", response.getStatusMessage());
+      }
+
+      return convertExceptionIfUnsuccessful(response).parseAs(Table.class);
+    } catch (IOException e) {
+      throw new RuntimeIOException("%s", e);
+    }
+  }
+
+  @Override
+  public void delete(TableReference tableReference) {
+    try {
+      load(tableReference); // Fetching it to validate it is a BigQuery Metastore table first
+
+      HttpResponse response =
+          client
+              .tables()
+              .delete(
+                  tableReference.getProjectId(),
+                  tableReference.getDatasetId(),
+                  tableReference.getTableId())
+              .executeUnparsed();
+
+      if (response.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+        throw new NoSuchTableException("%s", response.getStatusMessage());
+      }
+
+      convertExceptionIfUnsuccessful(response);
+    } catch (IOException e) {
+      throw new RuntimeIOException("%s", e);
+    }
+  }
+
+  @Override
+  public List<Tables> list(DatasetReference datasetReference, boolean filterUnsupportedTables) {
```

Comment:
This flag, `filterUnsupportedTables`, is inverted from the behavior of the `list-all-tables` property.  So, if you set list all, it actually filters the tables.